tests/test_multipart_operations.py

```diff
@@ -4,92 +4,282 @@
+
+        Test basic s3 operations using a noobaa bucket:
+        1. Create an account
+        2. Create a bucket using S3
+        3. Write multipart objects to the bucket
+        4. List multipart objects from the bucket
+
+        """
+        log.info("Uploading multipart object")
+        resp = self.upload_multipart_object(c_scope_s3client, tmp_directories_factory)
+        obj_name = resp["object_names"][0]
+        log.info("Trying to complete multipart operation for the object")
+        mp_response = c_scope_s3client.complete_multipart_object_upload(
+            resp["bucket_name"],
+            obj_name,
+            resp[f"{obj_name}" + "_upload_id"],
+            resp["all_part_info"],
         )
         assert (
             mp_response["ResponseMetadata"]["HTTPStatusCode"] == 200
         ), "Failed to upload multipart object"
         log.info(mp_response)
 
-    # 4. List multipart objects from the bucket
-    listed_objs = s3_client.list_objects(bucket_name)
-    assert set(object_names).issubset(
-        set(listed_objs)
-    ), "All uploaded objects are not present in bucket"
+    def test_s3_multipart_download(self, c_scope_s3client, tmp_directories_factory):
+        """
+
+        Test basic s3 operations using a noobaa bucket:
+        1. Write multipart objects to the bucket
+        2. Download the objects from the bucket and verify data integrity
+
+        """
+        log.info("Uploading multipart object")
+        resp = self.upload_multipart_object(c_scope_s3client, tmp_directories_factory)
+        obj_name = resp["object_names"][0]
+        log.info("Trying to complete multipart operation for the object")
+        mp_response = c_scope_s3client.complete_multipart_object_upload(
+            resp["bucket_name"],
+            obj_name,
+            resp[f"{obj_name}" + "_upload_id"],
+            resp["all_part_info"],
+        )
+        log.info(mp_response)
+        log.info("Trying to download multipart object and validating it")
+        c_scope_s3client.download_bucket_contents(
+            resp["bucket_name"], resp["results_dir"]
+        )
+        assert check_data_integrity(resp["origin_dir"], resp["results_dir"])
+        log.info("Both uploaded and downloaded data are identical")
+
+    def test_s3_multipart_object_list(self, c_scope_s3client, tmp_directories_factory):
+        """
+
+        Test multipart object list operations using BOTO s3:
+        1. Write multipart objects to the bucket
+        2. List objects from the bucket and verify it
+
+        """
+        log.info("Uploading multipart object")
+        resp = self.upload_multipart_object(c_scope_s3client, tmp_directories_factory)
+        obj_name = resp["object_names"][0]
+        log.info("Trying to complete multipart operation for the object")
+        mp_response = c_scope_s3client.complete_multipart_object_upload(
+            resp["bucket_name"],
+            obj_name,
+            resp[f"{obj_name}" + "_upload_id"],
```

Comment:
f"{obj_name}" is equivalent to obj_name - see the other comment for suggestoins

Reply:
Fixed